### PR TITLE
Handled adherent who forgot his password

### PR DESCRIPTION
**Pull request by HeahDude**
Closes [EM-2](http://jira.sensiolabs.net/browse/EM-2).

### CODE REVIEW

File: src/AppBundle/Controller/Security/AdherentSecurityController.php [hunk 1]
```diff
@@ -46,4 +55,70 @@ public function loginCheck()
     public function logoutAction()
     {
     }
+
+    /**
+     * @Route("/mot-de-passe-oublie", name="adherent_forgot_password")
+     * @Method("GET|POST")
+     */
+    public function forgotPasswordAction(Request $request)
```
Comment (hhamon) [inline src/AppBundle/Controller/Security/AdherentSecurityController.php]: Can you name the action with a verb please like `retrieveForgotPasswordAction()`.

File: src/AppBundle/Controller/Security/AdherentSecurityController.php [hunk 2]
```diff
@@ -46,4 +55,70 @@ public function loginCheck()
...
+            ->add('email', EmailType::class, ['constraints' => new NotBlank()])
+            ->add('submit', SubmitType::class)
+            ->getForm()
+        ;
+
+        if ($form->handleRequest($request)->isSubmitted() && $form->isValid()) {
+            $email = $form->get('email')->getData();
+            $adherent = $this->getDoctrine()->getRepository(Adherent::class)->findByEmail($email);
+
+            if ($adherent) {
```
Comment (hhamon) [inline src/AppBundle/Controller/Security/AdherentSecurityController.php]: `if ($adherent = $this->getDoctrine()->getRepository(Adherent::class)->findByEmail($email)) {`

File: tests/AppBundle/TestHelperTrait.php [hunk 3]
```diff
@@ -77,4 +84,32 @@ public function getMailjetEmailRepository(): MailjetEmailRepository
...
+     */
+    protected function createAdherent(string $email = null)
+    {
+        $phone = new PhoneNumber();
+        $phone->setCountryCode('FR');
+        $phone->setNationalNumber('0140998211');
+
+        return new Adherent(
+            Adherent::createUuid($email ?: 'john.smith@example.org'),
+            'john.smith@example.org',
```
Comment (hhamon) [inline tests/AppBundle/TestHelperTrait.php]: Should be `$email ?: 'john.smith@example.org'` as well.

File: src/AppBundle/Controller/Security/AdherentSecurityController.php (hunk 1 above)
Comment (HeahDude) [inline src/AppBundle/Controller/Security/AdherentSecurityController.php]: Ok, done.

File: src/AppBundle/Controller/Security/AdherentSecurityController.php (hunk 2 above)
Comment (HeahDude) [inline src/AppBundle/Controller/Security/AdherentSecurityController.php]: I hesitated, the line is very long now, but 👍

File: tests/AppBundle/TestHelperTrait.php (hunk 3 above)
Comment (HeahDude) [inline tests/AppBundle/TestHelperTrait.php]: Right, fixed.

**Comment by HeahDude** [comment]
Comments addressed.

**Pull request merged**